build.md

```diff
@@ -0,0 +1,109 @@
+# Amplifier Build Instructions
+
+This document provides instructions for building Amplifier components, including contracts and the `ampd` service.
+
+## Prerequisites
+
+- Rust and Cargo (rust-version = "1.81.0" currently required)
+- Docker (for contract optimization)
+- Git (for cloning and handling submodules)
+- cargo-run-script (for running optimization scripts)
+
+## Building the Project
+
+### Building Everything for Development
+
+To build the entire project for development and testing:
+
+```bash
+cargo build
+```
+
+This will compile all packages, contracts, and the ampd service for your local machine architecture.
+
+### Building Specific Components for Development
+
+To build a specific component for development/testing purposes:
+
+```bash
+cd contracts/<contract-name>
+cargo build
+```
+Or run tests:
+```bash
+cd contracts/<contract-name>
+cargo test
+```
+
+## Building and Optimizing Contracts for Deployment
+
+CosmWasm contracts need to be built for the WebAssembly target and optimized for deployment. The optimization process is essential for reducing contract size and gas costs.
+
+### Installing cargo-run-script
+
+First, install the cargo-run-script tool:
```

Comment:
```suggestion
Install the cargo-run-script tool:
```
implied by section header, otherwise keep the sentence and remove the section header. Try to keep things concise in general